src/devices/src/virtio/console/queue_handler.rs

```diff
@@ -0,0 +1,116 @@
+use crate::virtio::console::Error;
+use event_manager::{EventOps, Events, MutEventSubscriber};
+use log::error;
+use std::io::{stdin, Read, Stdout};
+use vm_memory::GuestAddressSpace;
+use vmm_sys_util::epoll::EventSet;
+use vmm_sys_util::eventfd::EventFd;
+
+use crate::virtio::console::inorder_handler::InOrderQueueHandler;
+use crate::virtio::SingleFdSignalQueue;
+
+const STDIN_EVENT: u32 = 0;
+const TRANSMITQ_EVENT: u32 = 1;
+const RECEIVEQ_EVENT: u32 = 2;
+
+const STDIN_BUFFER_SIZE: usize = 1024;
```

Comment:
Where is this taken from?

Reply:
This is just a random value. To figure out a good one we should run some tests. I tried with some python tests but I could not measure the time it takes to send data to the driver accurately.